plugins/net_plugin/net_plugin.cpp

```diff
@@ -1189,32 +1191,43 @@ namespace eosio {
          fc_ilog( logger, "completing enqueue_sync_block ${num} to ${p}", ("num", num)("p", peer_name()) );
       }
       connection_wptr weak = shared_from_this();
-      app().post( priority::medium, [num, weak{std::move(weak)}]() {
-         connection_ptr c = weak.lock();
-         if( !c ) return;
-         controller& cc = my_impl->chain_plug->chain();
-         signed_block_ptr sb;
-         try {
-            sb = cc.fetch_block_by_number( num );
-         } FC_LOG_AND_DROP();
-         if( sb ) {
-            c->strand.post( [c, sb{std::move(sb)}]() {
-               c->enqueue_block( sb, true );
-            });
-         } else {
-            c->strand.post( [c, num]() {
-               peer_ilog( c, "enqueue sync, unable to fetch block ${num}", ("num", num) );
-               c->send_handshake();
-            });
-         }
-      });
+      connection_ptr c = weak.lock();
+      if (!c){
+          return true;
+      }
```

Comment:
No need to convert from weak pointer to shared_ptr as this is fine to use on the net_plugin thread.